### PR TITLE
Alternate Ubuntu CI fix

### PR DESCRIPTION
**Pull request by WillAyd**
@MarcoGorelli @lithomas1 @jbrockmendel 

I think if we clean up some of this dead code we can support a clean build on newer gcc. Lots of layers of indirection to sift through so not entirely clear on what's happening, but I think gcc isn't going to be happy with some of the paths this theoretically produces

https://github.com/pandas-dev/pandas/issues/22234

### CODE REVIEW

File: .github/workflows/32-bit-linux.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ permissions:
 
 jobs:
   pytest:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
```
Comment (lithomas1) [inline .github/workflows/32-bit-linux.yml:20]: Please don't unpin this. You should change this to ubuntu-22.04 everywhere.

Comment (WillAyd) [reply .github/workflows/32-bit-linux.yml:20]: 20.04 or 22.04? This was just a git revert

Comment (lithomas1) [reply .github/workflows/32-bit-linux.yml:20]: We can bump up to 22.04 if CI is green.

Comment (mroeschke) [reply .github/workflows/32-bit-linux.yml:20]: xref https://github.com/pandas-dev/pandas/issues/50115

**Review by MarcoGorelli — APPROVED** [review]
Seems to be working, nice!

**Comment by phofl** [comment]
This is also failing on 1.5.x, can we backport or should we pin the ubuntu version?

**Comment by MarcoGorelli** [comment]
probably fine to backport?

**Comment by MarcoGorelli** [comment]
@meeseeksdev please backport to 1.5.x

**Pull request merged**